### ISSUE TITLE
Yandex Disk support for 

### ISSUE THREAD

**Issue by JonasDralle** [duplicati/duplicati#1918]
I have:
- [x] searched open and closed issues for duplicates

----------------------------------------

### Version info

**Duplicati Version:** 2.0.1.20_canary_2016-09-01
**Operating System:** Windows 10

I would like some support for Yandex Disk.
It's a russian cloud hoster who supports YesREST, WebDAV and SDK.

I am using it myself but because Yandex is really huge in russia i though that

- adding it would be easy due to WebDAV
- adding it would put a smile on the face of some russian users (+ increase in russian users)

This is a (small) feature request
Website of Yandey Disk https://disk.yandex.com/
API https://tech.yandex.com/disk/

[![Image](https://i.imgur.com/Gg5vVai.png)](https://tech.yandex.com/disk/)

**Comment by JonasDralle** [comment]
I don't know if this is appropiate. I would not benefit from you supporting Yandex Disk (basically Google Drive for russians).
Feel free to close this issue if you are not interested.

**Comment by renestach** [comment]
Duplicati supports WebDAV already. Would you simply add it to the list of supported backends and pre-fill the server address?

**Comment by kenkendk** [comment]
From what I can read, you need to support the OAuth flow, and then you can opt to use WebDAV instead of their custom REST API.

**Comment by JonasDralle** [comment]
@renestach that's basically what i meant. It's easy but rewarding

**Comment by JonasDralle** [comment]
besides that i totally misread the sentence and you would need to implement OAuth flow
Sorry for that

**Comment by vnau** [comment]
Yandex.Disk supports WebDAV using login and password without OAuth flow. I've tested Duplicati 2.0.1.53_experimental_2017-03-13 with Yandex.Disk via WebDAV backend and it works like a charm.

Adding predefined list including webdav.yandex.com:443 server in WebDAV configuration would ease backup setup for Yandex.Disk users (~16 mln).

**Comment by parfilov** [comment]
+1 to Yandex Disc

**Comment by JonasDralle** [comment]
I have reworded the ticket. Yandex seems to be a mediocre cloud provider.
WebDAV is limited to maximum of 2 GB file size (instead of 50 GB when using the SDK) but this seems negible.

**Comment by parfilov** [comment]
@JonasDralle I've sent a ticket to the support. Maybe they'll do something with SDK. If not, it's more simple and cheaper to use Backblaze for me.

**Comment by JonasDralle** [comment]
which support?

**Comment by parfilov** [comment]
@JonasDralle Yandex

**Comment by avmaksimov** [comment]
> 

No. It's normal cloud provider but WebDAV was the feature before. Now it's unofficial feature and YaDisk support says that. Also Yandex slows down the speed for WebDAV, so we need an official support via Yandex API.

**Comment by vnau** [comment]
> Since October 18, 2019, Yandex.Disk has started throttling WebDAV, causing some clients to time out, the official page redirects to the Yandex.Disk 3.0 client page, support writes: “**Yandex.Disk is a personal service that is not designed to be used as an infrastructure element. For such tasks, we have Yandex.Cloud**"

I do not think that it is reasonable to implement support for the Native API, because existing limitations of WebDAV is a deliberate policy of Yandex and there is no guarantee that Native API would not be throttled like they did with WebDAV.

**Comment by avmaksimov** [comment]
> > Since October 18, 2019, Yandex.Disk has started throttling WebDAV, causing some clients to time out, the official page redirects to the Yandex.Disk 3.0 client page, support writes: “**Yandex.Disk is a personal service that is not designed to be used as an infrastructure element. For such tasks, we have Yandex.Cloud**"
> 
> I do not think that it is reasonable to implement support for the Native API, because existing limitations of WebDAV is a deliberate policy of Yandex and there is no guarantee that Native API would not be throttled like they did with WebDAV.

I use Yandex Native API in another program for several years and there are no problems with it.

So if somebody would do it I can give some money for bear (starting from 100$))).
